src/controllers/oracle.controller.ts

```diff
@@ -0,0 +1,265 @@
+        },
+      },
+      '401': {
+        description: 'Unauthorized',
+        content: {
+          'application/json': {
+            schema: {
+              $ref: '#/components/schemas/ErrorResponse',
+            },
+          },
+        },
+      },
+      '404': {
+        description: 'Oracle Not Found',
+        content: {
+          'application/json': {
+            schema: {
+              $ref: '#/components/schemas/ErrorResponse',
+            },
+          },
+        },
+      },
+    },
+    requestBody: {
+      content: {
+        'application/json': {
+          schema: {
+            $ref: '#/components/schemas/GoodOracle_Input',
+          },
+          examples: {
+            'Create an Oracle': {
+              value: {
+                name: 'Leyline',
+                goodOracleURI: 'leyline.gg',
+              },
+            },
+          },
+        },
+      },
+      description: '',
+    },
+    description: 'Create a new Oracle',
+    parameters: [],
+    security: [
+      {
+        Oracle_API_Key: [],
+      },
+    ],
+  })
+  async postOracle(
+    @requestBody({
+      content: {
+        'application/json': {
+          schema: {
+            $ref: '#/components/schemas/GoodOracle_Input',
+          },
+          examples: {
+            'Create an Oracle': {
+              value: {
+                name: 'Leyline',
+                goodOracleURI: 'leyline.gg',
+              },
+            },
+          },
+        },
+      },
+      description: '',
+    })
+    oracle: GoodOracle,
+  ): Promise<unknown> {
+    oracle.id = 0;
+    return this.goodOracleRepository.create(oracle);
+  }
+
+  /**
+   * Retrieve a Proof of Good Oracle
+   *
+   * @param id The PoG ID of the oracle
+   * @returns A Proof of Good Oracle
+   */
+  @operation('get', '/oracle/{id}', {
+    summary: 'Get Oracle',
+    operationId: 'get-oracle',
+    responses: {
+      '200': {
+        description: 'A Proof of Good Oracle',
+        content: {
+          'application/json': {
+            schema: {
+              $ref: '#/components/schemas/GoodOracle',
+            },
+            examples: {},
+          },
+        },
+      },
+      '404': {
+        description: 'Oracle Not Found',
+        content: {
+          'application/json': {
+            schema: {
+              $ref: '#/components/schemas/ErrorResponse',
+            },
+          },
+        },
+      },
+    },
+    description: 'Retrieve a Proof of Good Oracle',
+    parameters: [
+      {
+        schema: {
+          type: 'integer',
+          example: 3,
+        },
+        name: 'id',
+        in: 'path',
+        required: true,
+        description: 'The PoG ID of the oracle',
+      },
+    ],
+  })
+  async getOracle(
+    @param({
+      schema: {
+        type: 'integer',
+        example: 3,
+      },
+      name: 'id',
+      in: 'path',
+      required: true,
+      description: 'The PoG ID of the oracle',
+    })
+    id: number,
```

Comment:
```suggestion
    @param({
      schema: {
        type: 'string',
        example: "kOX9jdrNJjiRW72O1fLk",
      },
      name: 'id',
      in: 'path',
      required: true,
      description: 'The PoG ID of the oracle',
    })
    id: string,
  ): Promise<GoodOracle> {
```

as per your changes to id being a string